app/slave/cluster_slave.py

```diff
@@ -56,6 +58,29 @@ def __init__(self, port, host, num_executors=10):
         self._build_teardown_coin = None
         self._base_executor_index = None
 
+        self._heartbeat_count = 0
+        self._heartbeat_count_threshold = Configuration['heartbeat_count_threshold']
+        self._heartbeat_frequency = Configuration['heartbeat_frequency']
+        self.scheduler = BackgroundScheduler()
+        self._heartbeat_job = None
+        self._logger.info('Heartbeat will run every {} seconds'.format(self._heartbeat_frequency))
+
+    def configure_heartbeat(self):
+        self._heartbeat_job = self.scheduler.add_job(self.heartbeat, 'interval', seconds=self._heartbeat_frequency)
+        return self.scheduler
+
+    def heartbeat(self):
+        state_url = self._master_api.url('slave', self._slave_id, 'heartbeat')
+        try:
+            self._network.post_with_digest(state_url, request_params={'slave': {'heartbeat': True}},
```

Comment:
This is unfortunate. I'd really like to see this be
`yield self._network.post_with_digest(...)`

Unfortunately post_with_digest doesn't return a future. And rewriting it to return a future would be pretty involved. Definitely doesn't belong in this pr. Can you add a TODO to

1. Rewrite network.post_with_digest to return a future/be a coroutine
2. Run this code on tornado's event loop instead of on it's own thread